Tests/DependencyInjection/MonologExtensionTest.php

```diff
@@ -186,6 +187,97 @@ public function testExceptionWhenUsingGelfWithoutPublisherHostname()
         $loader->load([['handlers' => ['gelf' => ['type' => 'gelf', 'publisher' => []]]]], $container);
     }
 
+    public function testExceptionWhenUsingLegacyGelfImplementationWithUnsupportedEncoder(): void
+    {
+        if (!class_exists('Gelf\MessagePublisher')) {
+            class_alias(DummyClassForClassExistsCheck::class, 'Gelf\MessagePublisher');
```

Comment:
Doing such class aliases is a hack, and won't work properly. If the `testExceptionWhenUsingGelfWithInvalidEncoder` test is run first, this test will never reach the code path guarded by `class_exists('Gelf\MessagePublisher')` (because the actual guard to reach that code path is `!class_exists('Gelf\Transport\UdpTransport') && class_exists('Gelf\MessagePublisher')` and you cannot remove the other alias).

Tests should be written against the installed dependencies, without such hacks, and skipped in case they are not relevant.

Reply:
Thanks for the feedback; I completely agree. While this approach is hackish, it did feel wrong that the changes couldn’t be properly tested. I initially used these as part of my testing process and I left them in the PR...

As I mentioned in my previous comment:

> Generally, workarounds involving class existence checks are hacky, but I wanted to ensure the tests provided meaningful coverage.

I'll remove them.